gnucash/gnome-utils/gnc-cell-renderer-text-flag.c

```diff
@@ -0,0 +1,222 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, contact:
+ *
+ * Free Software Foundation           Voice:  +1-617-542-5942
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org
+ */
+
+#include <config.h>
+#include <string.h>
+#include <gtk/gtk.h>
+
+#include "gnc-cell-renderer-text-flag.h"
+
+/**
+ * @Title: GncCellRendererTextFlag
+ *
+ * A #GncCellRendererTextFlag extends the GtkCellRendererText
+ * adding the avility to show a color triangle on the top right corner
+ * to flag the cell.
+ */
+
+
+static void gnc_cell_renderer_text_flag_get_property(GObject *object,
+                                                     guint param_id,
+                                                     GValue *value,
+                                                     GParamSpec *pspec);
+static void gnc_cell_renderer_text_flag_set_property(GObject *object,
+                                                     guint param_id,
+                                                     const GValue *value,
+                                                     GParamSpec *pspec);
+
+static void gnc_cell_renderer_text_flag_render(
+    GtkCellRenderer *cell, cairo_t *cr, GtkWidget *widget,
+    const GdkRectangle *background_area, const GdkRectangle *cell_area,
+    GtkCellRendererState flags);
+
+enum
+{
+    PROP_0,
+
+    PROP_FLAG_SIZE,
+    PROP_FLAG_COLOR,
+    PROP_FLAGGED,
+
+    LAST_PROP
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GncCellRendererTextFlag,
+                           gnc_cell_renderer_text_flag,
+                           GTK_TYPE_CELL_RENDERER_TEXT)
+
+static void
+gnc_cell_renderer_text_flag_init(GncCellRendererTextFlag *celltext)
+{
+    GncCellRendererTextFlagPrivate *priv;
+    GtkCellRendererText *cell = GTK_CELL_RENDERER_TEXT(celltext);
+
+    celltext->priv =
+        gnc_cell_renderer_text_flag_get_instance_private(celltext);
+    priv = celltext->priv;
+
+    priv->size = 8;
+    gdk_rgba_parse(&priv->color, "red");
+    priv->flagged = FALSE;
+}
+
+static void
+gnc_cell_renderer_text_flag_class_init(GncCellRendererTextFlagClass *class)
+{
+    GObjectClass *object_class       = G_OBJECT_CLASS(class);
+    GtkCellRendererClass *cell_class = GTK_CELL_RENDERER_CLASS(class);
+
+    object_class->get_property = gnc_cell_renderer_text_flag_get_property;
+    object_class->set_property = gnc_cell_renderer_text_flag_set_property;
+
+    cell_class->render = gnc_cell_renderer_text_flag_render;
+
+    g_object_class_install_property(
+        object_class, PROP_FLAG_SIZE,
+        g_param_spec_int("flag-size", "Flag size", "Flag size", 0, 50,
+                         8, G_PARAM_READWRITE));
+
+    g_object_class_install_property(
+        object_class, PROP_FLAG_COLOR,
+        g_param_spec_string("flag-color", "Flag color name",
+                            "Flag color as a string", "red",
```

Comment:
While this will replace the hardcoded "beige" colour from a previous commit, the basic idea remains the same: don't hardcode the colour. Make the colour a css/styling property of this widget. That way users can override it via css.

Reply:
I leave this part with a default "red", and configure in the view from css